pkg/agent/secondarynetwork/podwatch/controller.go

```diff
@@ -200,36 +201,114 @@ func (pc *PodController) processCNIUpdate(e interface{}) {
 	if !ok {
-		// NOTE: We do not handle Pod annotation deletion/update scenario at present.
 		klog.V(2).InfoS("Pod does not have a NetworkAttachmentDefinition", "Pod", klog.KObj(pod))
-		return nil
+	} else {
+		// Parse Pod annotation and proceed with the secondary network configuration.
+		var err error
+		networkList, err = netdefutils.ParseNetworkAnnotation(secondaryNetwork, pod.Namespace)
+		if err != nil {
+			klog.ErrorS(err, "Error when parsing network annotation", "annotation", secondaryNetwork)
+			// Do not return an error as a retry is not appropriate.
+			// When the annotation is fixed, the Pod will be enqueued again.
+			return nil
+		}
 	}
-	// Parse Pod annotation and proceed with the secondary network configuration.
-	networklist, err := netdefutils.ParseNetworkAnnotation(secondaryNetwork, pod.Namespace)
-	if err != nil {
-		klog.ErrorS(err, "Error when parsing network annotation", "annotation", secondaryNetwork)
-		// Do not return an error as a retry is not appropriate.
-		// When the annotation is fixed, the Pod will be enqueued again.
+
+	if len(storedSecondaryInterfaces) > 0 {
+		if len(networkList) > 0 {
+			// We do not support secondary network update at the moment. Return as long as one
+			// secondary interface has been created for the Pod.
+			klog.V(1).InfoS("Secondary network already configured on this Pod. Changes to secondary network configuration are not supported, skipping update",
+				"Pod", klog.KObj(pod))
+			return nil
+		}
+		if err := pc.removeInterfaces(storedSecondaryInterfaces); err != nil {
+			return err
+		}
+	}
+
+	var netStatus []netdefv1.NetworkStatus
+	var err error
+	if len(networkList) > 0 {
+		netStatus, err = pc.configurePodSecondaryNetwork(pod, networkList, podCNIInfo)
+		if err != nil {
+			return err
+		}
+	}
+
+	if netStatus != nil {
+		_ = pc.updatePodNetworkStatusAnnotation(netStatus, pod)
+	} else {
+		_ = pc.deletePodNetworkStatusAnnotation(pod)
+	}
+	return nil
+}
+
+func (pc *PodController) deletePodNetworkStatusAnnotation(pod *corev1.Pod) error {
+	resultErr := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		pod, err := pc.kubeClient.CoreV1().Pods(pod.Namespace).Get(context.TODO(), pod.Name, metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+		if _, ok := pod.Annotations[netdefv1.NetworkStatusAnnot]; ok {
+			delete(pod.Annotations, netdefv1.NetworkStatusAnnot)
+			_, err = pc.kubeClient.CoreV1().Pods(pod.Namespace).UpdateStatus(context.TODO(), pod, metav1.UpdateOptions{})
+			return err
+		}
 		return nil
+	})
+	if resultErr != nil {
+		klog.ErrorS(resultErr, "Pod network status annotation delete failed", "Pod", klog.KObj(pod))
+		return fmt.Errorf("status delete failed for pod %s/%s: %v", pod.Namespace, pod.Name, resultErr)
+	}
+	klog.V(2).InfoS("Pod network status annotation deleted", "Pod", klog.KObj(pod))
+	return nil
+}
+
+var (
+	netdefutilsSetNetworkStatus = netdefutils.SetNetworkStatus
```

Comment:
Probably just name the func variable: setNetworkStatus

Reply:
done